jcl/src/java.base/share/classes/openj9/management/ThreadInfoBase.java

```diff
@@ -0,0 +1,322 @@
+		threadState = Thread.State.values()[state];
+		suspended = isSuspended;
+		inNative = isInNative;
+		blockedCount = blockedCountVal;
+		blockedTime = blockedTimeVal;
+		waitedCount = waitedCountVal;
+		waitedTime = waitedTimeVal;
+
+		if (blockingObjectOwner != null) {
+			lockOwnerId = blockingObjectOwner.getId();
+			lockOwnerName = blockingObjectOwner.getName();
+		} else {
+			lockOwnerId = -1;
+			lockOwnerName = null;
+		}
+
+		stackTrace = stackTraceVal;
+		String lockNameTemp = null;
+		if (blockingObject != null) {
+			lockNameTemp = blockingObject.getClass().getName() + '@'
+					+ Integer.toHexString(System.identityHashCode(blockingObject));
+		}
+
+		lockedMonitors = lockedMonitorsVal;
+		lockedSynchronizers = lockedSynchronizersVal;
+
+		if (blockingObject != null) {
+			blockingLockInfo = new LockInfoBase(blockingObject.getClass().getName(),
+					System.identityHashCode(blockingObject));
+		} else {
+			blockingLockInfo = null;
+		}
+		if (null != blockingLockInfo) {
+			lockNameTemp = blockingLockInfo.toString();
+		}
+		lockName = lockNameTemp;
+	}
+
+	/**
+	 * @return a text description of this thread info.
+	 */
+	@Override
+	public String toString() {
+		// Since ThreadInfos are immutable the string value need only be
+		// calculated the one time
+		String ls = System.lineSeparator();
+		if (TOSTRING_VALUE == null) {
+			StringBuilder result = new StringBuilder();
+/*[IF Java11]*/
+			result.append(String.format("\"%s\" prio=%d Id=%d %s", //$NON-NLS-1$
+					threadName, Integer.valueOf(priority), Long.valueOf(threadId), threadState));
+/*[ELSE]*/
+			result.append(String.format("\"%s\" Id=%d %s", //$NON-NLS-1$
+					threadName, Long.valueOf(threadId), threadState));
+/*[ENDIF]*/
+			if (State.BLOCKED == threadState) {
+				result.append(String.format(" on %s owned by \"%s\" Id=%d", //$NON-NLS-1$
+						lockName, lockOwnerName, Long.valueOf(lockOwnerId)));
+			}
+			result.append(ls);
+			if ((stackTrace != null) && (stackTrace.length > 0)) {
+				MonitorInfoBase[] lockArray = new MonitorInfoBase[stackTrace.length];
+				for (MonitorInfoBase mi : lockedMonitors) {
+					int lockDepth = mi.stackDepth;
+					lockArray[lockDepth] = mi;
+				}
+				int stackDepth = 0;
+				for (StackTraceElement element : stackTrace) {
+					result.append("\tat "); //$NON-NLS-1$
+					Util.printStackTraceElement(element, null, result, true);
+					result.append(ls);
+					if (null != lockArray[stackDepth]) {
+						MonitorInfoBase mi = lockArray[stackDepth];
+						result.append(String.format("\t- locked %s%n", //$NON-NLS-1$
+								mi.toString()));
+					}
+					stackDepth += 1;
+				}
+			}
+			TOSTRING_VALUE = result.toString();
+		}
+		return TOSTRING_VALUE;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj) {
+			return true;
+		}
+
+		boolean result = (obj != null)  && (obj instanceof ThreadInfoBase);
+		// Safe to cast if the instanceof test passed.
+		ThreadInfoBase ti = (ThreadInfoBase) obj;
```

Comment:
No, the cast is _not_ safe because it occurs even if the instanceof test yields false.

Reply:
True.  Thanks.